hazelcast/src/main/java/com/hazelcast/cluster/impl/ClusterServiceImpl.java

```diff
@@ -1285,7 +1293,7 @@ private void sendMembershipEventNotifications(MemberImpl member, Set<Member> mem
             final MembershipServiceEvent event = new MembershipServiceEvent(membershipEvent);
             for (final MembershipAwareService service : membershipAwareServices) {
                 // service events should not block each other
-                nodeEngine.getExecutionService().execute(ExecutionService.SYSTEM_EXECUTOR, new Runnable() {
+                nodeEngine.getExecutionService().execute(MEMBERSHIP_EVENT_EXECUTOR_NAME, new Runnable() {
```

Comment:
is there any reason why it did use `SYSTEM_EXECUTOR` originally? The Event executor can be quite busy (specially when you use listeners, etc..) and it's dropping events when a queue is full - that's consider a business as usual. Are we ok if we drop a membership event? Can't this make the whole cluster unstable?

I just checked source code and apparently SYSTEM_EXECUTOR has capacity defined as `Integer.MAX_VALUE`.